core/includes/image.inc

```diff
@@ -585,6 +598,53 @@ function image_add_svg_attributes($svg_content, array $attributes) {
   return $doc->saveXML($doc->firstChild);
 }
 
+/**
+ * Get an SVG image's defined dimensions.
+ *
+ * @param string $uri
+ *   The URI or path to an SVG image.
+ *
+ * @return array|FALSE
+ *   An array containing the keys "width" and "height" as integer values. If the
+ *   image is an SVG but no set dimensions are available, these keys will have
+ *   NULL values. If the image is not an SVG, or the SVG file is empty or
+ *   invalid, FALSE will be returned.
+ */
+function image_get_svg_dimensions($uri) {
+  // Safety check.
+  if (!image_is_svg($uri)) {
+    return FALSE;
+  }
+
+  // Return cached dimensions if already retrieved once this request.
+  $cached_images = &backdrop_static(__FUNCTION__, array());
+  if (isset($cached_images[$uri])) {
+    return $cached_images[$uri];
+  }
+
+  // If there are no dimensions, store the dimensions as zero.
+  // Required for compatibility with image field storage.
+  $svg_dimensions = array(
+    'width' => 0,
+    'height' => 0,
```

Comment:
Why zeroes? That's the least useful value IMO. :wink:

Reply:
The idea is that if it can't find dimensions on the SVG it'll default to zero. I tried null but that caused a fatal error when saving an image field, since the image field stores the height and width as integers.

Reply:
I got, that this "0" is to have something to save to the database. :+1: 

However, this value's quite problematic for image_styles (resizing doesn't happen, as "0" is always smaller than any image style setting will be).
Another concern: using that as fallback forces us to override later, anyway. We can never use that value as-is ... Not helpful, right? :wink:

Reply:
The constants are back!